src/dsp/BandSplitter/BandSplitterProcessor.cpp

```diff
@@ -73,7 +95,19 @@ void BandSplitterProcessor::processBlock (const chowdsp::BufferView<const float>
             processFilter (bandFilters.filter12);
     };
 
-    if (params.threeBandOnOff->get())
+    if (params.fourBandOnOff->get())
+    {
+        bufferMid.clear();
```

Comment:
I think we also need to clear the `lowMid` and `highMid` buffers in the three-band and two-band cases.